intel/IA-Hardware-Composer
# Pre-built shader support (NEW)

This pull request replaces the previous pull request (#292).

Now there are three changes under this pull request, which basically make build process now takes care of pre-building of shaders automatically if provided this new configuration option:

--prebuilt-shader-pci-id=<PCI ID in hex number form (e.g 0x1234) >

AND

shader-db library and executable (intel_stub.so, run) exist under common/compositor/gl/gl_shader_pre_built/shader-db/

Once binary data (shader program binary) is generated, it will be added to the HWC source as header files and compiled together with it.

glprogram.cpp tries to load this static data first. If it does not exist or fail to load it, it tries to get load this binary data from files at /vendor/etc if available. Then it goes to run-time compilation path if none of those prebuilt shaders are available or functional.

Jira: None.
Test: 
A. built in binary test

0. Place <shader-db> project under ./common/compositor/gl/gl_shader_pre_built/

1. Build HWC:
./autogen.sh --prefix=$WLD --prebuilt-shader-pci-id=<current GPU's PCI-ID>
make && make install

2. Verify testlayer works with built-in pre-built shaders

B. shader program files test

1. Build HWC:
  ./autogen.sh --prefix=$WLD
  make && make install

2. run ./common/compositor/gl/gl_shader_pre_built/generate_shader_test.sh
    use shader-db to build shader program binaries from generated shader-test files 
    move *.shader_test.bin over to under /vendor/etc
   
3. Verify testlayer works with built-in pre-built shaders

Signed-off-by: Dongwon Kim dongwon.kim@intel.com
